src/main/java/org/thoughtcrime/securesms/ProfileAdapter.java

```diff
@@ -0,0 +1,372 @@
+    memberCount = memberList!=null ? memberList.length : 0;
+
+    itemData.add(new ItemData(ITEM_AVATAR, null, 0));
+
+    if (isSelfTalk || dcContact != null && !dcContact.getStatus().isEmpty()) {
+      itemDataStatusText = isSelfTalk ? context.getString(R.string.saved_messages_explain) : dcContact.getStatus();
+      itemData.add(new ItemData(ITEM_SIGNATURE, itemDataStatusText, 0));
+    } else {
+      itemData.add(new ItemData(ITEM_DIVIDER, null, 0));
+    }
+
+    itemData.add(new ItemData(ITEM_ALL_MEDIA_BUTTON, context.getString(R.string.apps_and_media), R.drawable.ic_apps_24));
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      itemData.add(new ItemData(ITEM_SEND_MESSAGE_BUTTON, context.getString(R.string.send_message), R.drawable.ic_send_sms_white_24dp));
+    }
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      long lastSeenTimestamp = dcContact.getLastSeen();
+      String lastSeenTxt;
+      if (lastSeenTimestamp == 0) {
+        lastSeenTxt = context.getString(R.string.last_seen_unknown);
+      }
+      else {
+        lastSeenTxt = context.getString(R.string.last_seen_at, DateUtils.getExtendedTimeSpanString(context, lastSeenTimestamp));
+      }
+      itemData.add(new ItemData(ITEM_LAST_SEEN, lastSeenTxt, 0));
+    }
+
+    if (memberList!=null) {
+      itemData.add(new ItemData(ITEM_DIVIDER, null, 0));
+      if (dcChat != null) {
+        if (!isMailingList && dcChat.canSend()) {
+          itemData.add(new ItemData(ITEM_MEMBERS, DcContact.DC_CONTACT_ID_ADD_MEMBER, 0));
+          if (!isBroadcast) {
+            itemData.add(new ItemData(ITEM_MEMBERS, DcContact.DC_CONTACT_ID_QR_INVITE, 0));
+          }
+        }
+      }
+      for (int value : memberList) {
+        itemData.add(new ItemData(ITEM_MEMBERS, value, 0));
+      }
+    }
+
+    if (sharedChats != null && !isDeviceTalk) {
+      itemData.add(new ItemData(ITEM_HEADER, context.getString(R.string.profile_shared_chats), 0));
+      for (int i = 0; i < sharedChats.getCnt(); i++) {
+        itemData.add(new ItemData(ITEM_SHARED_CHATS, 0, i));
+      }
+    }
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      int verifierId = dcContact.getVerifierId();
+      if (verifierId != 0) {
+        String introducedBy;
+        if (verifierId == DcContact.DC_CONTACT_ID_SELF) {
+          introducedBy = context.getString(R.string.verified_by_you);
+        } else {
+          introducedBy = context.getString(R.string.verified_by, dcContext.getContact(verifierId).getDisplayName());
+        }
+        itemData.add(new ItemData(ITEM_INTRODUCED_BY, introducedBy, 0));
+      }
+
+      if (dcContact != null) {
+        itemData.add(new ItemData(ITEM_ADDRESS, dcContact.getAddr(), 0));
+      } else if (isMailingList) {
+        itemData.add(new ItemData(ITEM_ADDRESS, dcChat.getMailinglistAddr(), 0));
+      }
+    }
+
+    notifyDataSetChanged();
+  }
+
+  public int ALL_MEDIA_COUNT_MAX = 500;
```

Comment:
why 500 and not 999 or 99? to me 99+ would already be good enough, not sure a higher precision number brings much utility here compared to potential performance cost of the calculation